### PR TITLE
Add Data Source Elasticsearch Domain

### LINKED ISSUE #7899: data aws_elasticsearch_domain support
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Terraform should support `data "aws_elasticsearch_domain"` to get attributes for an existing domain.

### New or Affected Resource(s)

* aws_elasticsearch_domain

### Potential Terraform Configuration

```hcl
data "aws_elasticsearch_domain" "example" {
  name = "example-domain"
}
```

### PR DESCRIPTION
**Pull request by kterada0509**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes #7899 

Changes proposed in this pull request:

- [x] Add `aws_elasticsearch_domain` data source
- [x] Add `aws_elasticsearch_domain` data source acceptance test
- [x] Add `aws_elasticsearch_domain` data source document

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccDataSourceAWSElasticSearchDomain_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccDataSourceAWSElasticSearchDomain_ -timeout 120m
=== RUN   TestAccDataSourceAWSElasticSearchDomain_basic
=== PAUSE TestAccDataSourceAWSElasticSearchDomain_basic
=== RUN   TestAccDataSourceAWSElasticSearchDomain_withDedicatedMaster
=== PAUSE TestAccDataSourceAWSElasticSearchDomain_withDedicatedMaster
=== RUN   TestAccDataSourceAWSElasticSearchDomain_withPolicy
=== PAUSE TestAccDataSourceAWSElasticSearchDomain_withPolicy
=== RUN   TestAccDataSourceAWSElasticSearchDomain_withEncryptAtRestDefaultKey
=== PAUSE TestAccDataSourceAWSElasticSearchDomain_withEncryptAtRestDefaultKey
=== RUN   TestAccDataSourceAWSElasticSearchDomain_withNodeToNodeEncryption
=== PAUSE TestAccDataSourceAWSElasticSearchDomain_withNodeToNodeEncryption
=== RUN   TestAccDataSourceAWSElasticSearchDomain_complex
=== PAUSE TestAccDataSourceAWSElasticSearchDomain_complex
=== CONT  TestAccDataSourceAWSElasticSearchDomain_basic
=== CONT  TestAccDataSourceAWSElasticSearchDomain_withNodeToNodeEncryption
=== CONT  TestAccDataSourceAWSElasticSearchDomain_withPolicy
=== CONT  TestAccDataSourceAWSElasticSearchDomain_complex
=== CONT  TestAccDataSourceAWSElasticSearchDomain_withDedicatedMaster
=== CONT  TestAccDataSourceAWSElasticSearchDomain_withEncryptAtRestDefaultKey
--- PASS: TestAccDataSourceAWSElasticSearchDomain_withEncryptAtRestDefaultKey (780.75s)
--- PASS: TestAccDataSourceAWSElasticSearchDomain_basic (843.03s)
--- PASS: TestAccDataSourceAWSElasticSearchDomain_withPolicy (883.27s)
--- PASS: TestAccDataSourceAWSElasticSearchDomain_withNodeToNodeEncryption (903.07s)
--- PASS: TestAccDataSourceAWSElasticSearchDomain_complex (919.82s)
--- PASS: TestAccDataSourceAWSElasticSearchDomain_withDedicatedMaster (969.28s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	969.368s
```

### CODE REVIEW

File: website/docs/d/elasticsearch_domain.html.markdown [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+* `enabled` - Whether to enable node-to-node encryption. If the `node_to_node_encryption` block is not provided then this defaults to `false`.
+
+**vpc_options** the following attributes are exported:
+
+AWS documentation: [VPC Support for Amazon Elasticsearch Service Domains](https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-vpc.html)
+
+**Note** you must have created the service linked role for the Elasticsearch service to use the `vpc_options`.
+If you need to create the service linked role at the same time as the Elasticsearch domain then you must use `depends_on` to make sure that the role is created before the Elasticsearch domain.
+See the [VPC based ES domain example](#vpc-based-es) above.
```
Comment (longwave) [inline website/docs/d/elasticsearch_domain.html.markdown]: This looks to be copied from the resource and doesn't seem to make sense in the data context.

File: website/docs/d/elasticsearch_domain.html.markdown [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+* `advanced_options` - Key-value string pairs to specify advanced configuration options.
+   Note that the values for these configuration options must be strings (wrapped in quotes) or they
+   may be wrong and cause a perpetual diff, causing Terraform to want to recreate your Elasticsearch
+   domain on every apply.
+* `ebs_options` - EBS related options, may be required based on chosen [instance size](https://aws.amazon.com/elasticsearch-service/pricing/). See below.
+* `encrypt_at_rest` - Encrypt at rest options. Only available for [certain instance types](http://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/aes-supported-instance-types.html). See below.
+* `node_to_node_encryption` - Node-to-node encryption options. See below.
+* `cluster_config` - Cluster configuration of the domain, see below.
+* `snapshot_options` - Snapshot related options, see below.
+* `vpc_options` - VPC related options, see below. Adding or removing this configuration forces a new resource ([documentation](https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-vpc.html#es-vpc-limitations)).
```
Comment (longwave) [inline website/docs/d/elasticsearch_domain.html.markdown]: The second sentence is only relevant to a resource, not a data source.

File: website/docs/d/elasticsearch_domain.html.markdown [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+* `arn` - Amazon Resource Name (ARN) of the domain.
+* `domain_id` - Unique identifier for the domain.
+* `domain_name` - The name of the Elasticsearch domain.
+* `endpoint` - Domain-specific endpoint used to submit index, search, and data upload requests.
+* `kibana_endpoint` - Domain-specific endpoint for kibana without https scheme.
+* `vpc_options.0.availability_zones` - If the domain was created inside a VPC, the names of the availability zones the configured `subnet_ids` were created inside.
+* `vpc_options.0.vpc_id` - If the domain was created inside a VPC, the ID of the VPC.
+* `access_policies` - IAM policy document specifying the access policies for the domain
+* `advanced_options` - Key-value string pairs to specify advanced configuration options.
+   Note that the values for these configuration options must be strings (wrapped in quotes) or they
```
Comment (longwave) [inline website/docs/d/elasticsearch_domain.html.markdown]: This doesn't make sense in a data source context.

**Comment by jeffsteinmetz** [comment]
Does the data source implementation in this PR support wildcard searches, like domain = `vpc-dev-elasticsearch-*`?
similar to VPC, security group, etc data sources?

**Comment by ryndaniels** [comment]
Hi @kterada0509! 👋 Thanks so much for submitting this PR! Unfortunately, it looks to be a duplicate of #1867, and our policy is to review the oldest PR first. Sorry we didn't catch this earlier, but since we are moving forward with that, we hope to have this data source available soon!

**Pull request closed without merging**